src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs

```diff
@@ -113,7 +113,7 @@ internal partial class CodeFixService : ICodeFixService
             }
 
             var buildOnlyDiagnosticsService = document.Project.Solution.Services.GetRequiredService<IBuildOnlyDiagnosticsService>();
-            allDiagnostics.AddRange(buildOnlyDiagnosticsService.GetBuildOnlyDiagnostics(document.Id));
+            allDiagnostics = allDiagnostics.AddRange(buildOnlyDiagnosticsService.GetBuildOnlyDiagnostics(document.Id));
```

Comment:
@mavasani @sharwell @jaredpar Is there an analyzer somewhere to generally catch mistakes like this? I know I've heard plenty of discussions of it, but did anybody ever write one?

Reply:
@jasonmalinowski https://github.com/dotnet/runtime/issues/34098 tracks adding such an analyzer driven by a new attribute in the framework. This is supposed to be lot more refined than all our existing analyzers in same space (IDE0058, IDE0059, etc.).
Tagging @eerhardt @buyaa-n as the primary champions for the design of that analyzer. Not sure if this will make it into .NET8 as it seems to have large cost estimate.

Reply:
Note that we did have CA1806 which used to fire for the Collections.Immutable methods, but the `[Pure]` attribute was removed on these APIs in https://github.com/dotnet/runtime/pull/35118. The `CA1806` analyzer no longer fires because there are no `[Pure]` attributes on the APIs.

> Not sure if this will make it into .NET8 as it seems to have large cost estimate.

No, https://github.com/dotnet/runtime/issues/34098 has been pushed out of .NET 8.

cc @jeffhandley

Reply:
In the interim, it might make sense to modify CA1806 to explicitly include items we know have return values that shouldn't be dropped.